web/src/main/kotlin/App.kt

```diff
@@ -1,20 +1,20 @@
 import com.surrus.common.remote.Assignment
-import com.surrus.common.remote.PeopleInSpaceApi
 import react.*
 import react.dom.*
 import kotlinx.coroutines.*
 
 
+
 val App = functionalComponent<RProps> { _ ->
-    val scope = MainScope()
-    val api = PeopleInSpaceApi()
+    val appState = useContext(AppStateContext)
 
     val (people, setPeople) = useState(emptyList<Assignment>())
 
-    useEffect(dependencies = listOf()) {
-        scope.launch {
-            setPeople(api.fetchPeople().people)
+    useEffectWithCleanup(dependencies = listOf()) {
+        appState.mainScope.launch {
+            setPeople(appState.peopleInSpaceApi.fetchPeople().people)
         }
+        return@useEffectWithCleanup { appState.mainScope.cancel() }
```

Comment:
I'd lift the scope out of the app state and instead have a scope for this component. Right now this clean up would cancel all coroutines launched in that scope.
I'd use the useMemo hook to create a scope for this functional component